src/Features/Core/Portable/ConvertForToForEach/AbstractConvertForToForEachCodeRefactoringProvider.cs

```diff
@@ -194,6 +194,23 @@ bool IterationVariableIsUsedForMoreThanCollectionIndex(SyntaxNode current)
                             return true;
                         }
 
+                        // Avoid situations where list[i] is assigned:
+                        // list[i] = 1;
+                        // (list[i], a) = (1, 2);
+                        // list[i] += 1;
+                        // list[i]++;
+                        var c = current.Parent.Parent.Parent;
```

Comment:
this is pretty non-good.  no understanidng of why jumping up 3 parents is appropriate.

Reply:
@CyrusNajmabadi , please look some lines above for the same `current.Parent.Parent.Parent;`. BTW, as I understand, this is your code you wrote the last year  :-)

Reply:
My terrible coding patterns are no excuse ;-)   But i do take this back.  I missed that right above we checked all the intermediary states, so thsi is fine.  #resolved